apis/config/v1beta1/configuration_types.go

```diff
@@ -372,6 +373,30 @@ type ClusterQueueVisibility struct {
 type Resources struct {
 	// ExcludedResourcePrefixes defines which resources should be ignored by Kueue
 	ExcludeResourcePrefixes []string `json:"excludeResourcePrefixes,omitempty"`
+
+	// Transformations defines how to transform PodSpec resources into Workload resource requests.
+	// +listType=map
+	// +listMapKey=input
+	// +optional
+	Transformations []ResourceTransformation `json:"transformations,omitempty"`
+}
+
+type ResourceTransformationStrategy string
+
+const Retain ResourceTransformationStrategy = "Retain"
+const Replace ResourceTransformationStrategy = "Replace"
+
+type ResourceTransformation struct {
+	// Input is the name of the input resource.
+	Input corev1.ResourceName `json:"input"`
```

Comment:
Mark this as required to get CRD-level validation. And MinLength as 1.

Reply:
Allowing a length 0 output array in conjunction with a `Replace` directive enables an input resource to be removed.  (ie, a  non-fuzzy way of doing `excludeResourcePrefix` without unintended matches).

Reply:
Expanded the comment to explain use case for 0 length outputs.